### PR TITLE
CDAP-13088 Make compatible with new AuditMessage for Metadata

### PR DESCRIPTION
**Pull request by rohitsinha54**
Changes to make tracker compatible with: https://issues.cask.co/browse/CDAP-13088
PR in CDAP: https://github.com/caskdata/cdap/pull/10215

### CODE REVIEW

File: src/main/java/co/cask/tracker/entity/AuditLogTable.java [hunk 1]
```diff
@@ -197,8 +203,18 @@ private long getInvertedTsScanKeyPart(long time) {
    * @param row the row from the scanner to build the AuditMessage from
    * @return a new AuditMessage based on the information in the row
    */
-  private static AuditMessage createAuditMessage(Row row) {
-    EntityId entityId = GSON.fromJson(row.getString("entityId"), EntityId.class);
+  @VisibleForTesting
+  static AuditMessage createAuditMessage(Row row) {
+    // backward compatibility CDAP-13088
+    MetadataEntity metadataEntity;
+    String entity = row.getString("entityId");
```
Comment (anew) [inline src/main/java/co/cask/tracker/entity/AuditLogTable.java]: maybe call this variable "json" or something similar, instead of "entity".

File: src/main/java/co/cask/tracker/utils/DiscoveryMetadataClient.java [hunk 2]
```diff
@@ -186,6 +186,11 @@ protected URL resolve(NamespaceId namespace, String path) throws MalformedURLExc
     }
   }
 
+  @Override
+  protected URL resolve(String s) throws IOException {
```
Comment (anew) [inline src/main/java/co/cask/tracker/utils/DiscoveryMetadataClient.java]: why was this addition needed?

Comment (rohitsinha54) [reply src/main/java/co/cask/tracker/utils/DiscoveryMetadataClient.java]: In CDAP the MetadataClient now has this resolve which just takes path and not namespaceId because now we don't specify metadata entity specifically in namespace it is custom and can not have a namespace.
